e2e/test/Helpers/templates/FaultInjectionPoolingOverAmqp.cs

```diff
@@ -51,7 +51,7 @@ public static class FaultInjectionPoolingOverAmqp
             logger.Trace($">>> {nameof(FaultInjectionPoolingOverAmqp)} Initializing Device Clients for multiplexing test.");
             for (int i = 0; i < devicesCount; i++)
             {
-                TestDevice testDevice = await TestDevice.GetTestDeviceAsync(logger, $"{devicePrefix}_{i}_").ConfigureAwait(false);
+                TestDevice testDevice = await TestDevice.GetTestDeviceAsync(logger, $"{devicePrefix}_{Guid.NewGuid()}").ConfigureAwait(false);
```

Comment:
This should create unique devices across parallel tests, which should help fix the concurrency issues.

Reply:
this is just the prefix to the device Id that we send here, the index `i` is to help identify which of the multiplexed devices got faulted. 
`TestDevice.GetTestDeviceAsync()` implementation already appends a prefix to the device name, so uniqueness is automatically ensured.

This change needs to be reverted.

Reply:
hmmm. Then maybe I don't understand something here. How do we ensure we only check the amqpConnectionStatuses for the specific devices ? Or is that not per device? How can I ensure we only check the amqpConnectionStatuses for this run and set of devices?

Reply:
you use the list of test devices `testDevices`, which are unique to each test run.